spec/assignPRReviewersSpec.js

```diff
@@ -0,0 +1,155 @@
+
+describe('Assign PR Reviewers', () => {
+  /**
+   * @type {import('probot').Probot} robot
+   */
+  let robot;
+
+  /**
+   * @type {import('probot').Octokit} github
+   */
+  let github;
+
+  /**
+   * @type {import('probot').Application} app
+   */
+  let app;
+
+  beforeEach(() => {
+    spyOn(scheduler, 'createScheduler').and.callFake(() => {});
+
+    // Spy on other modules that will be triggered by the payload.
+    spyOn(apiForSheetsModule, 'checkClaStatus').and.callFake(() => {});
+    spyOn(
+      checkPullRequestLabelsModule,
+      'checkChangelogLabel'
+    ).and.callFake(() => {});
+    spyOn(checkWipDraftPRModule, 'checkWIP').and.callFake(() => {});
+    spyOn(checkPullRequestJobModule, 'checkForNewJob').and.callFake(() => {});
+    spyOn(checkPullRequestBranchModule, 'checkBranch').and.callFake(() => {});
+    spyOn(
+      checkCriticalPullRequestModule,
+      'checkIfPRAffectsDatastoreLayer'
+    ).and.callFake(() => {});
+
+    github = {
+      issues: {
+        addAssignees: jasmine.createSpy('addAssignees').and.resolveTo({}),
+        createComment: jasmine.createSpy('createComment').and.resolveTo({}),
+      },
+    };
+    robot = createProbot({
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiabot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(assignPRReviewersModule, 'assignAllCodeowners').and.callThrough();
+  });
+
+  describe('when a pull request is created', () => {
+    const codeowners = [
+      { login: 'reviewer1' },
+      { login: 'reviewer2' },
+      { login: 'reviewer3' },
+    ];
+
+    beforeEach(async () => {
+      pullRequestPayload.payload.pull_request.requested_reviewers = codeowners;
+      await robot.receive(pullRequestPayload);
+    });
+
+    it('should call assignAllCodeowners', () => {
+      expect(assignPRReviewersModule.assignAllCodeowners).toHaveBeenCalled();
+    });
+
+    it('should assign all codeowners', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        owner: pullRequestPayload.payload.repository.owner.login,
+        repo: pullRequestPayload.payload.repository.name,
+        issue_number: pullRequestPayload.payload.pull_request.number,
+        assignees: ['reviewer1', 'reviewer2', 'reviewer3'],
+      });
+    });
+
+    it('should ping codeowners', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+      expect(github.issues.createComment).toHaveBeenCalledWith({
+        owner: pullRequestPayload.payload.repository.owner.login,
+        repo: pullRequestPayload.payload.repository.name,
+        issue_number: pullRequestPayload.payload.pull_request.number,
+        body:
+          'Assigning @reviewer1, @reviewer2, @reviewer3 for the ' +
+          'first pass review of this PR. Thanks!',
+      });
+    });
+  });
+
+  describe('when a pull request has no codeowners', () => {
```

Comment:
no codeowners -> no codeowners other than the PR author

Reply:
Done.